pkg/parse/events/meta_scheduler_test.go

```diff
@@ -0,0 +1,233 @@
+				SyncPeriod: time.Second,
+			},
+			stepSize: time.Second,
+			expectedEvents: []eventResult{
+				{
+					Event:  SyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  SyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  SyncEvent{},
+					Result: HandleResult{},
+				},
+			},
+		},
+		{
+			name: "ResyncEvents From ResyncPeriod",
+			scheduler: &MetaScheduler{
+				FullSyncPeriod: time.Second,
+			},
+			stepSize: time.Second,
+			expectedEvents: []eventResult{
+				{
+					Event:  FullSyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  FullSyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  FullSyncEvent{},
+					Result: HandleResult{},
+				},
+			},
+		},
+		{
+			name: "StatusEvents From StatusUpdatePeriod",
+			scheduler: &MetaScheduler{
+				StatusUpdatePeriod: time.Second,
+			},
+			stepSize: time.Second,
+			expectedEvents: []eventResult{
+				{
+					Event:  StatusEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  StatusEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  StatusEvent{},
+					Result: HandleResult{},
+				},
+			},
+		},
+		{
+			name: "NamespaceResyncEvents From NamespaceControllerPeriod",
+			scheduler: &MetaScheduler{
+				NamespaceControllerPeriod: time.Second,
+			},
+			stepSize: time.Second,
+			expectedEvents: []eventResult{
+				{
+					Event:  NamespaceResyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  NamespaceResyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  NamespaceResyncEvent{},
+					Result: HandleResult{},
+				},
+			},
+		},
+		{
+			name: "RetryEvents From RetryBackoff",
+			scheduler: &MetaScheduler{
+				RetryBackoff: wait.Backoff{
+					Duration: time.Second,
+					Factor:   0, // no backoff, just 1s delay
+					Steps:    3, // at least as many as we expect
+				},
+			},
+			stepSize: time.Second,
+			expectedEvents: []eventResult{
+				{
+					Event:  RetrySyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  RetrySyncEvent{},
+					Result: HandleResult{},
+				},
+				{
+					Event:  RetrySyncEvent{},
+					Result: HandleResult{},
+				},
+			},
+		},
+		{
+			name: "RetryEvents From RetryBackoff with ResetRetryBackoff",
+			scheduler: &MetaScheduler{
+				RetryBackoff: wait.Backoff{
+					Duration: time.Second,
+					Factor:   0, // no backoff, just 1s delay
+					Steps:    1, // just one, to show the count resets when ResetRetryBackoff is true
+				},
+			},
+			stepSize: time.Second,
+			expectedEvents: []eventResult{
+				{
+					Event: RetrySyncEvent{},
+					Result: HandleResult{
+						ResetRetryBackoff: true,
+					},
+				},
+				{
+					Event: RetrySyncEvent{},
+					Result: HandleResult{
+						ResetRetryBackoff: true,
+					},
+				},
+				{
+					Event: RetrySyncEvent{},
+					Result: HandleResult{
+						ResetRetryBackoff: true,
+					},
+				},
+			},
+		},
+		// TODO: test retry backoff behavior
```

Comment:
Testing retry backoff may not be easy with the way this function is only exposing Events and Results. We need to be able to adjust the Step size between each event to trigger the next retry event.